crate_universe/src/context/crate_context.rs

```diff
@@ -650,23 +667,89 @@ impl CrateContext {
             .iter()
             .flat_map(|target| {
                 target.kind.iter().filter_map(move |kind| {
-                    // Unfortunately, The package graph and resolve graph of cargo metadata have different representations
-                    // for the crate names (resolve graph sanitizes names to match module names) so to get the rest of this
-                    // content to align when rendering, the package target names are always sanitized.
+                    // Unfortunately, The package graph and resolve graph of cargo metadata have
+                    // different representations for the crate names (resolve
+                    // graph sanitizes names to match module names) so to get the rest of this
+                    // content to align when rendering, the package target names are always
+                    // sanitized.
                     let crate_name = sanitize_module_name(&target.name);
 
-                    // Locate the crate's root source file relative to the package root normalized for unix
+                    // Locate the crate's root source file relative to the package root normalized
+                    // for unix
                     let crate_root = pathdiff::diff_paths(&target.src_path, package_root).map(
-                        // Normalize the path so that it always renders the same regardless of platform
+                        // Normalize the path so that it always renders the same regardless of
+                        // platform
                         |root| root.to_string_lossy().replace('\\', "/"),
                     );
+                    let mut srcs = Glob::new_rust_srcs().into();
+                    let mut crate_root = crate_root;
+                    let mut compile_data = None;
+                    let local_patch = package.id.repr.contains("(path+file://");
```

Comment:
Annoyingly cargo metadata doesn't give us the information we want (the patch path) in an easy to access way. The `source` field which should probably have it is just set to `null`, and the `id` field (which is supposed to be an opaque identifier) has something like `<name> <version> (path+file://<location>)`. This PR uses that `id` to determine whether a crate has been patched, and I'm fairly confident that if the cargo maintainers did actually change the format of the `id` field then that would mean they have the bandwidth to fix the `source` field so we'd still be able to get the info somehow.